### DISCUSSION TITLE
Validate Controller name on Array

### DISCUSSION THREAD

**Discussion by kkroeger93** [react-hook-form/react-hook-form#5198] [Q&A]
Is it possible to validate an input registered as an array ?

```
<Controller
      name="company[0].companyName"
      control={control}
      shouldUnregister={true}
      defaultValue={defaultValues.companyName}
      rules={{ required: true }}
      render={({ field }) => (
          <TextField
              variant="outlined"
              label="Firmenname"
              className={classes.input}
              {...field}
          />
      )}
/>
```

```
name="company[0].companyName"
```

This returns TypeError: Cannot read property '0' of undefined
```
{errors && console.log(errors.company[0])}
```

**Comment by bluebill1049** [comment]
you can use `validate` function, but you can only return a single error message or array of error messages.

**Comment by kkroeger93** [comment]
I'm using yup as a a resolver for validating.
What I like to do is passing an array e.g. 

`contact[0].firstName` 

Is there a way to work with such array ? 
FYI: I would like do this with arrays so there is a way to pre structure all my form fields into different" sections, e.g. contact[ ] or addresses[ ].. this makes it more easily for me to store data correctly. 


And could you maybe provide me an example on how to do so ? 

I could try to give you more information with an CSB if this would be more helpful.
